### DISCUSSION TITLE
Copying files from ServerFilesCourse copies the full path

### DISCUSSION THREAD

**Discussion by mfbutner** [PrairieLearn/PrairieLearn#6099] [General]
I used the ServerFilesCourse option for the first time on an External Graded question and I was surprised that when I specified a file to be copied, it copied the entire directory structure up to the file instead of just the file. For example if I had
```
"externalGradingOptions": {
    "serverFilesCourse": ["dirA/dirB/dirC/file.py"]
 }
```
Then in the grader image I would end up with `/grade/serverFilesCourse/dirA/dirB/dirC/file.py` when I expected I would end up with just `/grade/serverFilesCourse/file.py`. This behavior surprised me based on the documentation that I read on [External Grading](https://prairielearn.readthedocs.io/en/latest/externalGrading/). I assumed that if I specified a file/directory to be copied, then it would be copied directly into `/grade/serverFilesCourse`. Does it copy the whole path to workaround the issue that if multiple files with the same local name were copied they would overwrite each other? I.E. if some had done
```
"externalGradingOptions": {
    "serverFilesCourse": ["dirA/dirB/dirC/file.py", "dirA/file.py"]
 }
```

**Comment by echuber2** [comment]
Yes, there are good reasons for PL to not merge the included directories. Is this a problem for your grading software? If you know the filename is unique, you can search the included directories for it. Otherwise you may need to add some context clue in your autograder configuration so that it uses the right file from the right subdirectory.

**Reply by mfbutner** [reply]
It wasn't an issue I couldn't workaround once I figured out what was happening. I just had to figure out that this was the expected behavior first. I'll try and update the documentation to clarify that this is what will happen so others don't end up confused like I did.

**Comment by jonatanschroeder** [comment]
There are cases where keeping the directory structure is important, e.g., Java classes.

**Reply by mfbutner** [reply]
I would have thought that in those cases the copy would have been specified to start at the directory where the structure started to matter. I'm not trying to say that I'm right or that things should be changed. Just explain what I was thinking that lead to me thinking things were going to happen differently.
